operator/pkg/reconciliation/handler.go

```diff
@@ -27,6 +30,56 @@ import (
 // Use a var so we can mock this function
 var setControllerReference = controllerutil.SetControllerReference
 
+// key: Node.Name, value: CassandraDatacenter.Name
+var nodeToDcName = make(map[string]string)
+
+// key: Node.Name, value: CassandraDatacenter.Namespace
+var nodeToDcNamespace = make(map[string]string)
+
+// Get the dcName and dcNamespace for a node
+func DatacenterForNode(nodeName string) (string, string) {
+	dcName, ok := nodeToDcName[nodeName]
```

Comment:
I spent some time thinking about that, but I figured the mutex wouldn't be necessary because only one thread would ever be writing to this thing at a time.  I typically use mutex's to avoid multiple writers.  The reader is doing a check for key existence, which should either return true or false.  If a key is found, then a reconcile request is triggered, if not, nothing happens.

If a key hasn't been properly removed, then we suffer a needless reconcile call.  
If a key hasn't been properly added, then we miss reporting a node change event.  However, I find this situation highly unlikely, because we don't actually care about node events until our pods are being created, in which case we have probably gone through a few rounds of reconcile loops that properly updated this map.

In any case, I don't mind adding a mutex to be extra safe.  I just didn't see a pressing need for it.

Reply:
Looks like maps are not safe for concurrent read/write:

https://stackoverflow.com/questions/36167200/how-safe-are-golang-maps-for-concurrent-read-write-operations

If I understand the comments there correctly, it used to be okay to have a concurrent read with a write, so long as there were not concurrent writers, but that is no longer the case. I still don't know if the read and write are actually happening in separate threads though.

Reply:
TIL.  Ok, i'll wrap it in a concurrency primitive.